addons/website_slides/static/src/xml/slide_quiz.xml

```diff
@@ -45,28 +45,43 @@
     </t>
 
     <t t-name="slide.slide.quiz.validation">
-        <div t-if="widget.readonly" class="alert alert-info">
-            <b>Join course to take quiz</b>
-            <span class="my-0 h4" style="line-height: 1">
-                <span title="Succeed and gain karma" aria-label="Succeed and gain karma" class="badge badge-pill badge-warning text-white font-weight-bold ml-3 px-2 py-1">
-                    + <t t-esc="widget.quiz.quizKarmaGain"/> XP
-                </span>
-            </span>
-        </div>
-        <div t-else="" class="d-flex align-items-center justify-content-between">
-            <div t-att-class="'d-flex align-items-center' + (widget.slide.completed ? ' alert alert-success my-0 py-1 px-3' : '')">
-                <button t-if="! widget.slide.completed" role="button" title="Check answers" aria-label="Check answers"
-                    class="btn btn-primary text-uppercase font-weight-bold o_wslides_js_lesson_quiz_submit">Check your answers</button>
-                <b t-else="" class="my-0 h5">Done !</b>
-                <span class="my-0 h5" style="line-height: 1">
-                    <span role="button" title="Succeed and gain karma" aria-label="Succeed and gain karma" class="badge badge-pill badge-warning text-white font-weight-bold ml-3 px-2">
-                        + <t t-if="! widget.slide.completed" t-esc="widget.quiz.quizKarmaGain"/><t t-else="" t-esc="widget.quiz.quizKarmaWon"/> XP
+        <div id="validation">
+            <div t-if="widget.readonly &amp;&amp; !widget.publicUser &amp;&amp; widget.channel.enroll == 'public'" class="alert alert-info d-flex align-items-center justify-content-between o_wslides_course_join_widget">
+                <div class="o_wslides_js_course_join">
+                    <b>Join the course to take the quiz</b>
+                    <span class="my-0 h4" style="line-height: 1">
+                        <span title="Succeed and gain karma" aria-label="Succeed and gain karma" class="badge badge-pill badge-warning text-white font-weight-bold ml-3 px-2 py-1">
+                            + <t t-esc="widget.quiz.quizKarmaGain"/> XP
+                        </span>
+                    </span>
+                </div>
+                <!-- Here comes the Join button rendered by the widget -->
```

Comment:
I would put the button to replace the "Join the course to take the quiz" sentence. No need to have a string + a button, I think we can keep just a button.